hyperspy/io_plugins/hdf5.py

```diff
@@ -94,44 +94,86 @@ def get_hspy_format_version(f):
-                "HyperSpy hdf5 file format. I will attempt to load it, but, "
-                "if I fail, it is likely that I will be more successful at "
-                "this and other tasks if you upgrade me.")
-
-        experiments = []
-        exp_dict_list = []
-        if 'Experiments' in f:
-            for ds in f['Experiments']:
-                if isinstance(f['Experiments'][ds], h5py.Group):
-                    if 'data' in f['Experiments'][ds]:
-                        experiments.append(ds)
-            if not experiments:
-                raise IOError(not_valid_format)
-            # Parse the file
-            for experiment in experiments:
-                exg = f['Experiments'][experiment]
-                exp = hdfgroup2signaldict(exg)
-                exp_dict_list.append(exp)
-        else:
-            raise IOError('This is not a valid HyperSpy HDF5 file. '
-                          'You can still load the data using a hdf5 reader, '
-                          'e.g. h5py, and manually create a Signal. '
-                          'Please, refer to the User Guide for details')
-        return exp_dict_list
+                backing_store=False, load_to_memory=True, **kwds):
+    f = h5py.File(filename, mode=mode, driver=driver, **kwds)
+    # Getting the format version here also checks if it is a valid HSpy
+    # hdf5 file, so the following two lines must not be deleted or moved
+    # elsewhere.
+    global current_file_version
+    current_file_version = get_hspy_format_version(f)
+    global default_version
+    if current_file_version > default_version:
+        warnings.warn(
+            "This file was written using a newer version of the "
+            "HyperSpy hdf5 file format. I will attempt to load it, but, "
+            "if I fail, it is likely that I will be more successful at "
+            "this and other tasks if you upgrade me.")
+
+    experiments = []
+    exp_dict_list = []
+    if 'Experiments' in f:
+        for ds in f['Experiments']:
+            if isinstance(f['Experiments'][ds], h5py.Group):
+                if 'data' in f['Experiments'][ds]:
+                    experiments.append(ds)
+                    d = f['Experiments'][ds]['data']
+        if not experiments:
+            raise IOError(not_valid_format)
+        # Parse the file
+        for experiment in experiments:
+            exg = f['Experiments'][experiment]
+            exp = hdfgroup2signaldict(exg, load_to_memory)
+            exp_dict_list.append(exp)
+    else:
+        raise IOError('This is not a valid HyperSpy HDF5 file. '
+                      'You can still load the data using a hdf5 reader, '
+                      'e.g. h5py, and manually create a Signal. '
+                      'Please, refer to the User Guide for details')
+    if load_to_memory:
+        f.close()
+    return exp_dict_list
+
+
+# not used at the moment, but might be useful for the future
+def get_signal_chunks(data, metadata=None):
```

Comment:
If this chunk of code is not used currently I would suggest not to include it until it is actually used for something.